pages/language-tour.md

```diff
@@ -467,6 +459,13 @@ type Matrix3 {
 }
 ```
 
+{% comment %}
+I don't know if we discussed this, but making the memberwise init public has API resilience
+implications.  In particular, it means you can't simply add/remove stored properties, even if they
+were non-public, without breaking client code.  I'd like to design some facilities for helping a
+library author to ensure API stability eventually.
+{% endcomment %}
```

Comment:
I thought about that.

The idea is that if you want to change the layout of the type later on, you'll have to write a custom constructor to keep your promise. That's what you have to do in Swift today anyway. The memberwise init feature is just a shortcut for the a large class of situations where your type does not cross a resilience boundary but also doesn't want to expose all its parts publicly.

Reply:
Yes!  But the problem in Swift today is that AFAIK there's nothing in place to discourage you from changing the layout of your type once you've made the promise.  Actually in the case of API it's not necessarily about layout, just about which members are stored.